File: newsletter/src/main/java/org/com/newsletter/auth/domain/Member.java
package org.com.newsletter.auth.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.com.newsletter.time.BaseTime;

/**
 * 회원 엔티티.
 * - JPA 프록시 호환을 위해 기본 생성자는 PROTECTED
 * - createdAt/updatedAt 은 BaseTime(Instant) 사용.
 */
@Entity
@Table(name = "members", indexes = {
    @Index(name = "idx_member_username", columnList = "username", unique = true)
})
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
public class Member extends BaseTime {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 100)
    private String username;

    @Column(nullable = false)
    private String password;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Role role;

}

File: ssr-board/src/main/kotlin/org/com/ssrboard/domain/Member.kt
package org.com.ssrboard.domain

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EntityListeners
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime

@Entity
@EntityListeners(AuditingEntityListener::class)
class Member protected constructor(
    @Column(nullable = false, unique = true, length = 50)
    var nickname: String
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null

    @CreatedDate
    lateinit var createdAt: LocalDateTime
        private set

    @LastModifiedDate
    lateinit var updatedAt: LocalDateTime
        private set

    companion object {
        fun create(nickname: String) = Member(nickname)
    }
}